zookeeper-server/src/main/java/org/apache/zookeeper/server/quorum/FollowerRequestProcessor.java

```diff
@@ -115,6 +124,14 @@ public void run() {
         LOG.info("FollowerRequestProcessor exited loop!");
     }
 
+    private void maybeSendRequestToNextProcessor(Request request) throws RequestProcessorException {
+        if (request.isFromLearner() && skipLearnerRequestToNextProcessor) {
```

Comment:
Maybe you can invert the order, make the test over the field as first,
This option is disabled by default and we can save a bunch of CPU cycles

Reply:
That's a nice suggestion, @mayank99fb please consider this it.

Reply:
done